src/sfctl/tests/help_text_test.py

```diff
@@ -0,0 +1,364 @@
+
+        Returns the section name if the given line designates the beginning of a new section.
+        Returns None if the line does not. """
+
+        if (line.strip() and (line[0] != ' ')):
+            # Use these lines to set the 'section' variable and move on to the next line
+            line = line.strip().rstrip(':')
+            if line == 'Commands':
+                return 'Commands'
+            elif (line == 'Arguments') or (line == 'Global Arguments'):
+                return 'Arguments'
+            elif line == 'Group':
+                return 'Group'
+            elif line == 'Subgroups':
+                return 'Subgroups'
+            elif line == 'Command':
+                return 'Command'
+        return None
+
+    def validate_output(self, command_input, subgroups=(), commands=()): #pylint: disable=too-many-locals
+        """
+        This function verifies that the returned help text is correct, and that no exceptions
+        are thrown during invocation. It verifies correctness of help text by the following:
+
+        - The group has a description
+        - All listed subgroups and commands appear in order. We do not check for the
+            existence of extra subgroups and commands.
+        - If subgroups or commands is not provided, then we expect it not to appear in
+            the help text. If it does, there will be an assertion raised in this test.
+        - All listed subgroups and commands have descriptive text
+        - All listed arguments and global arguments have descriptive text.
+
+        If commands are provided, this function will call itself recursively to
+        verify the correctness of the commands, so that you do not need add another
+        entry to test for it.
+
+        Limitations: This test doesn't search for new commands which are added.
+                     If a test entry is not added here, then that entry will not be
+                     verified.
+                     While we can read in the lines to make this test more automated,
+                     such that no subcommands or groups need to be listed, that case
+                     will not catch whether or not a command or subgroups has been
+                     accidentally removed.
+
+        command_input (string): This represents the command for which you want to get the help text.
+            For example, "sfctl" or "sfctl application" or "sfctl application list".
+            Parameter command_input should not include the "-h" to get the help text, as this
+            method will take care of that.
+
+        subgroups (tuple of strings): This represents all of the subgroups expected in the
+            help text. This tuple must be in alphabetical order.
+
+        commands (tuple of strings): This represents all of the commands expected in the
+            help text. This tuple must be in alphabetical order.
+
+        Help text has one of the two following formats:
+
+        1.  Group
```

Comment:
This is way too long for a docstring. If you want to add this many details, best to do it in the Wiki.

Reply:
I agree that this is way too long, but this information is only relevant to this test, not suitable for the wiki I think. Unless you mean create a test reference section in the wiki and provide a link to it?

Reply:
Yeah I would rather do that. Look at how other modules do this, I feel like the docstrings are always short and then the readthedocs sections contain all the details.